### ISSUE TITLE
tidb-lightning segfault on macOS (tidb backend)

### ISSUE THREAD

**Issue by kolbe** [pingcap/br#1121]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

```
$ cat t/test.t1.0.csv
Year;Make;Model;Length
1997;Ford;E350;2,35
2000;Mercury;Cougar;2,38

$ /Users/kolbe/.tiup/components/tidb-lightning/v5.0.1/tidb-lightning --backend tidb --tidb-host 127.0.0.1 --tidb-port 4000 --tidb-user root --no-schema -d /Users/kolbe/t/
```

2. What did you expect to see?

3. What did you see instead?

```
Verbose debug logs will be written to /var/folders/_1/hv8fy2sx5qs3f2qkj7cfm4rm0000gn/T/lightning.log.2021-05-19T15.56.55-0700

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x5f3f80f]

goroutine 1 [running]:
github.com/pingcap/br/pkg/lightning/restore.LoadSchemaInfo(0x6d0e980, 0xc000369780, 0xc0000da0f8, 0x1, 0x1, 0xc0008ff248, 0x0, 0x0, 0xc0008ff258)
        github.com/pingcap/br@/pkg/lightning/restore/tidb.go:259 +0x1df
github.com/pingcap/br/pkg/lightning/restore.(*Controller).restoreSchema(0xc0006fb7a0, 0x6d0e980, 0xc000369780, 0xc0004f80f0, 0xc000746180)
        github.com/pingcap/br@/pkg/lightning/restore/restore.go:625 +0x3dc
github.com/pingcap/br/pkg/lightning/restore.(*Controller).Run(0xc0006fb7a0, 0x6d0e980, 0xc000369780, 0x1, 0x1)
        github.com/pingcap/br@/pkg/lightning/restore/restore.go:318 +0x243
github.com/pingcap/br/pkg/lightning.(*Lightning).run(0xc0005f3e00, 0x6d0e9c0, 0xc0000ca000, 0xc000923c00, 0x0, 0x0, 0x0, 0x0)
        github.com/pingcap/br@/pkg/lightning/lightning.go:315 +0xd77
github.com/pingcap/br/pkg/lightning.(*Lightning).RunOnce(0xc0005f3e00, 0x6d0e9c0, 0xc0000ca000, 0xc000923c00, 0x0, 0x0, 0x667b313, 0x11)
        github.com/pingcap/br@/pkg/lightning/lightning.go:191 +0xe9
main.main.func2(0xc000689b00, 0xc0005f3e00, 0x0, 0xc0006fc000)
        github.com/pingcap/br@/cmd/tidb-lightning/main.go:86 +0xa8
main.main()
        github.com/pingcap/br@/cmd/tidb-lightning/main.go:87 +0x57d
```

4. What version of BR and TiDB/TiKV/PD are you using?

```
$ /Users/kolbe/.tiup/components/tidb-lightning/v5.0.1/tidb-lightning -V
Release Version: v5.0.1
Git Commit Hash: b3bd5495369244cfb1d3f2c49763a7fdb55d4ecd
Git Branch: HEAD
Go Version: go1.13.4
UTC Build Time: 2021-04-24 12:57:15
Race Enabled: false

Release Version: v5.0.1
Edition: Community
Git Commit Hash: 1145e347d3469d8e89f88dce86f6926ca44b3cd8
Git Branch: heads/refs/tags/v5.0.1
UTC Build Time: 2021-04-23 05:50:09
GoVersion: go1.13.4
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false
```

5. Operation logs

tidb-lightning "verbose" log:
```
[2021/05/19 15:56:55.868 -07:00] [INFO] [info.go:40] ["Welcome to TiDB-Lightning"] [release-version=v5.0.1] [git-hash=b3bd5495369244cfb1d3f2c49763a7fdb55d4ecd] [git-branch=HEAD] [go-version=go1.13.4] [utc-build-time="2021-04-24 12:57:15"] [race-enabled=false]
[2021/05/19 15:56:55.869 -07:00] [INFO] [lightning.go:218] [cfg] [cfg="{\"id\":1621465015868766000,\"lightning\":{\"table-concurrency\":4,\"index-concurrency\":4,\"region-concurrency\":4,\"io-concurrency\":5,\"check-requirements\":true},\"tidb\":{\"host\":\"127.0.0.1\",\"port\":4000,\"user\":\"root\",\"status-port\":10080,\"pd-addr\":\"\",\"sql-mode\":\"ONLY_FULL_GROUP_BY,NO_AUTO_CREATE_USER\",\"tls\":\"false\",\"security\":{\"ca-path\":\"\",\"cert-path\":\"\",\"key-path\":\"\",\"redact-info-log\":false},\"max-allowed-packet\":67108864,\"distsql-scan-concurrency\":15,\"build-stats-concurrency\":20,\"index-serial-scan-concurrency\":20,\"checksum-table-concurrency\":2},\"checkpoint\":{\"schema\":\"tidb_lightning_checkpoint\",\"driver\":\"file\",\"enable\":true,\"keep-after-success\":false},\"mydumper\":{\"read-block-size\":65536,\"batch-size\":107374182400,\"batch-import-ratio\":0,\"data-source-dir\":\"file:///Users/kolbe/t\",\"character-set\":\"auto\",\"csv\":{\"separator\":\",\",\"delimiter\":\"\\\"\",\"null\":\"\\\\N\",\"header\":true,\"trim-last-separator\":false,\"not-null\":false,\"backslash-escape\":true},\"max-region-size\":268435456,\"filter\":[\"*.*\",\"!mysql.*\",\"!sys.*\",\"!INFORMATION_SCHEMA.*\",\"!PERFORMANCE_SCHEMA.*\",\"!METRICS_SCHEMA.*\",\"!INSPECTION_SCHEMA.*\"],\"files\":null,\"no-schema\":true,\"case-sensitive\":false,\"strict-format\":false,\"default-file-rules\":true},\"tikv-importer\":{\"addr\":\"\",\"backend\":\"tidb\",\"on-duplicate\":\"replace\",\"max-kv-pairs\":4096,\"send-kv-pairs\":32768,\"region-split-size\":100663296,\"sorted-kv-dir\":\"\",\"disk-quota\":0,\"range-concurrency\":0,\"engine-mem-cache-size\":536870912,\"local-writer-mem-cache-size\":134217728},\"post-restore\":{\"checksum\":\"off\",\"analyze\":\"off\",\"level-1-compact\":false,\"post-process-at-last\":true,\"compact\":false},\"cron\":{\"switch-mode\":\"5m0s\",\"log-progress\":\"5m0s\",\"check-disk-quota\":\"0s\"},\"routes\":null,\"security\":{\"ca-path\":\"\",\"cert-path\":\"\",\"key-path\":\"\",\"redact-info-log\":false},\"black-white-list\":{\"do-tables\":null,\"do-dbs\":null,\"ignore-tables\":null,\"ignore-dbs\":null}}"]
[2021/05/19 15:56:55.876 -07:00] [INFO] [lightning.go:285] ["load data source start"]
[2021/05/19 15:56:55.877 -07:00] [INFO] [loader.go:299] ["[loader] file is filtered by file router"] [path=.test.t1.0.csv.swp]
[2021/05/19 15:56:55.877 -07:00] [INFO] [lightning.go:288] ["load data source completed"] [takeTime=663.091µs] []
[2021/05/19 15:56:55.881 -07:00] [INFO] [restore.go:312] ["the whole procedure start"]
[2021/05/19 15:56:55.881 -07:00] [INFO] [tidb.go:331] ["skipping check requirements for tidb backend"]
```

tidb-server general log:
```
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET sql_mode='ONLY_FULL_GROUP_BY,NO_AUTO_CREATE_USER'"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET allow_auto_random_explicit_insert=1"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_opt_write_row_id=1"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_build_stats_concurrency=20"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_distsql_scan_concurrency=15"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_index_serial_scan_concurrency=20"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_checksum_table_concurrency=2"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET NAMES utf8mb4"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET sql_mode='ONLY_FULL_GROUP_BY,NO_AUTO_CREATE_USER'"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_checksum_table_concurrency=2"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET allow_auto_random_explicit_insert=1"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_opt_write_row_id=1"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_build_stats_concurrency=20"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_distsql_scan_concurrency=15"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET tidb_index_serial_scan_concurrency=20"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SET NAMES utf8mb4"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="SELECT variable_value FROM mysql.tidb WHERE variable_name = 'new_collation_enabled'"]
 [current_db=] [txn_mode=PESSIMISTIC] [sql="START TRANSACTION"]
9438867] [isReadConsistency=false] [current_db=] [txn_mode=PESSIMISTIC] [sql="SELECT version()"]
9438867] [isReadConsistency=false] [current_db=] [txn_mode=PESSIMISTIC] [sql="   SELECT table_name, column_name, column_type, extra    FROM information_schema.columns    WHERE table_schema = ?    ORDER BY table_name, ordinal_position [arguments: test]"]
9438867] [isReadConsistency=false] [current_db=] [txn_mode=PESSIMISTIC] [sql=COMMIT]
```

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `topology.yml` if deployed by TiUP

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus if possible

**Comment by IANTHEREAL** [comment]
can we close it now @Little-Wallace

**Comment by glorv** [comment]
closed by #1114
